### PR TITLE
Upgraded tokio, warp, hyper and actix

### PR DESCRIPTION
**Pull request by engylemure**
### Upgraded to

- `tokio = 1`
- `warp = 0.3`
- `hyper = 0.14`
- `actix = 4.0.0-beta.5`

### CODE REVIEW

**Comment by toxeus** [comment]
@LegNeato can we get this merged? The failing tests are not caused by this PR but originate from changes in the nightly compiler.

**Comment by dyedgreen** [comment]
I tried this locally, and it can basically be rebased with a one line change to the Cargo.toml on the current release

**Comment by toxeus** [comment]
Nice! This is rebased now and could be merged with a single click by someone that has sufficient perms.

As usual, the failing tests are triggered by changes in the nightly compiler and are unrelated to this PR.

**Comment by HGHimself** [comment]
I am having issues compiling `juniper_rocket` and I am curious if this is the same issue. Recently started happening, error message as follows:
```
error[E0277]: `?` couldn't convert the error to `(rocket::http::Status, std::string::String)`
   --> /Users/hgking/.cargo/registry/src/github.com-1ecc6299db9ec823/juniper_rocket-0.6.3/src/lib.rs:280:81
    |
280 |             .map_err(|e| Err((Status::InternalServerError, format!("{:?}", e))))?;
    |                                                                                 ^ the trait `std::convert::From<Result<_, (rocket::http::Status, std::string::String)>>` is not implemented for `(rocket::http::Status, std::string::String)`
    |
    = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
    = help: the following implementations were found:
              <(usize, usize) as std::convert::From<toml::tokens::Span>>
    = note: required because of the requirements on the impl of `FromResidual<Result<Infallible, Result<_, (rocket::http::Status, std::string::String)>>>` for `Outcome<GraphQLRequest<S>, (rocket::http::Status, std::string::String), rocket::Data>`
    = note: required by `from_residual`
```

**Comment by engylemure** [comment]
@HGHimself 
I'm not sure when this happened but probably `rocket` recently changed his API for the Error, removing some previous `From` implementations that they had, so this was causing the problem that you described above, this is unrelated to the problems that we are having on the nightly tests at the moment.

**Comment by toxeus** [comment]
@HGHimself I tried to pin down the rocket dependency in https://github.com/graphql-rust/juniper/pull/916 but the maintainer prefers the CI to repeatedly break and closed the PR :man_shrugging:

**Comment by TilBlechschmidt** [comment]
Without the intention of sounding condescending or pessimistic I'd like to add a little bit of personal opinion:

Tokio 1.0 has been released last Christmas and it has been over six months since then. By now, we are at version `1.6`! Since it was a breaking change with no runtime backwards compatibility, it is not possible to use `juniper` with *any* project that is using the latest Tokio runtime (and in turn anybody who is just getting started and wants to try something out using the current version of tokio without understanding the versioning hell).

Additionally, this has not been the first issue (#851  #852) or PR (#928 #912 #888 #910 #882 #849) on the topic. It makes it a hard sell to rely on this project long-term if *critical* issues like this one are ignored or stuck for such a long time.

Do not get me wrong, this project is amazing and a blaze to use. However, the fact that we are literally unable to use it in any "modern" project<sup>1</sup> with no official word on the issue is literally a deal-breaker. I totally understand that maintainers usually have other topics on their plate and from time to time do not have the capacity to deal with larger issues in their open-source projects. I've been guilty of that too, most people have I suspect. **However**, there is no harm in saying so with a friendly comment and/or clarifying how the community (us) can help push critical issues forward without much intervention from the maintainers.

From my perspective at least, almost all the work for the dependency upgrade madness to Tokio 1.0 and related packages has been done. What is left is a confirmation from the maintainers and a merge with a following release, or some feedback on what is causing headaches, issues or conflicts here.

In other words: Dear Maintainers, please comment on the issue with your gripes or merge the darn thing already.

<sub><sup>1</sup>The only way to use juniper is to basically force downgrade ***almost all*** your dependencies to a version that is over half a year old because most of them depend on the async runtime<sub>

**Comment by kunjee17** [comment]
@engylemure it seems actix is updated in master. I don't know how to give PR on PR but that conflict surely can be solved.

**Comment by LegNeato** [comment]
Thank you for the PR! Sorry it took so long to land. Will cut a release in the next day or so after I fix up the Rocket bit.

**Comment by samuela** [comment]
@LegNeato Would it be possible to also cut a new release for `juniper_hyper`? At the moment, the latest published version still uses an old version of hyper: https://docs.rs/crate/juniper_hyper/0.7.1.

**Comment by LegNeato** [comment]
This is now released on crates.io.
